### PR TITLE
ES6 dropdowns

### PR DESCRIPTION
**Pull request by regiskuckaertz**

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/ui/dropdowns.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        if (container) {
+            fastdom.write(() => {
+                container.classList.toggle('dropdown--active');
+                updateAria(container);
+            });
+        }
+    });
+};
+
+export { init };
```
Comment (sndrs) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: was nothing calling `updateAria`?

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: nope, haven't found any occurence 🤔

File: static/src/javascripts/projects/common/modules/ui/dropdowns.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import fastdom from 'fastdom';
+
+const containerSelector = '.dropdown';
+const buttonCN = 'dropdown__button';
+const contentCN = 'dropdown__content';
+
+const updateAria = (container: Element): void => {
+    const v: boolean = container.classList.contains('dropdown--active');
+    const content = [...container.getElementsByClassName(contentCN)];
+    const button = [...container.getElementsByClassName(buttonCN)];
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: Are you just using spread on DOM selectors as a standard? It doesn't look like there's any need to do this for the const `button` as we're applying the spread operator on it line 18 at `[...content, ...button]`

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: line 18 joins the two arrays, unless there's another syntax for that?

Comment (sndrs) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: that'll coerce them wont it?

Comment (sndrs) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: screen shot 2017-07-18 at 16 03 05

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: so if `content = [a, b]` and `button = [c, d]` then `[...content, ...button] = [a, b, c, d]`

I'm not sure I understand the question 😅

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: I mean it's just `content.concat(button)`. to be frank, I'm not sure it is needed (e.g. there might be only one content and one button), but bonzo abstracts on arrays, so this version is functionally equivalent.

Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: Here's an example of what I was referring to in my initial comment: https://jsbin.com/qomevaxosi/edit?html,js,console

Spread unpacks the values from the array and the node list into a single array.

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: oooh I see, the first spread is redundant. good catch 👍

File: static/src/javascripts/projects/common/modules/ui/dropdowns.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+const buttonCN = 'dropdown__button';
+const contentCN = 'dropdown__content';
+
+const updateAria = (container: Element): void => {
+    const v: boolean = container.classList.contains('dropdown--active');
+    const content = [...container.getElementsByClassName(contentCN)];
+    const button = container.getElementsByClassName(buttonCN);
+
+    content.forEach((c: Element) => {
+        c.setAttribute('aria-hidden', v ? 'false' : 'true');
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: You could do `.setAttribute('aria-hidden', !v)` and remove the conditional ternary operator here as the the value of `!v` will be converted to a string by `setAttribute`.

**Review by GHaberis — APPROVED** [review]
This looks good to me, I left one comment regarding `setAttribute` but it's just an alternative way of doing things.

File: static/src/javascripts/projects/common/modules/ui/dropdowns.js (hunk 3 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/ui/dropdowns.js]: Good call! What happens is that the boolean is coerced into a string before the call to `setAttribute`. To keep Flow happy, I'm doing this explicitly.

**Pull request merged**